src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs

```diff
@@ -542,14 +542,19 @@ internal static bool IsNullableEnabled(CSharpCompilation compilation)
             return options.IsFeatureEnabled(MessageID.IDS_FeatureStaticNullChecking);
         }
 
-        internal static void VerifyNoNullability(Symbol symbol)
+        internal static void VerifyUsesOfNullability(Symbol symbol, ImmutableArray<string> expectedUsesOfNullable)
         {
             var builder = ArrayBuilder<Symbol>.GetInstance();
             UsesIsNullableVisitor.GetUses(builder, symbol);
-            var symbols = builder.ToImmutableAndFree();
-            // Use AssertEx.Equal(Empty, symbols) rather than Assert.True(symbols.IsEmpty)
-            // so  the exception message includes the list of symbols.
-            AssertEx.Equal(ImmutableArray<Symbol>.Empty, symbols);
+
+            var format = SymbolDisplayFormat.TestFormat
+                .WithCompilerInternalOptions(SymbolDisplayCompilerInternalOptions.IncludeNonNullableTypeModifier | SymbolDisplayCompilerInternalOptions.IncludeNullableTypeModifier)
+                .RemoveParameterOptions(SymbolDisplayParameterOptions.IncludeName);
+
+            var symbols = builder.SelectAsArray(s => s.ToDisplayString(format));
+            builder.Free();
+
+            AssertEx.Equal(expectedUsesOfNullable, symbols, itemInspector: s => $"\"{s}\"");
```

Comment:
>AssertEx.Equal(expectedUsesOfNullable, symbols, itemInspector: s => $"\"{s}\""); [](start = 12, length = 80)

Is this check order sensitive? We might want it to be insensitive to the order.

Reply:
Yes, it is order-sensitive. I'll leave as-is for now. We can always relax later if it is inconvenient in practice.

---
In reply to: [183186989](https://github.com/dotnet/roslyn/pull/25928#discussion_r183186989) [](ancestors = 183186989)